draft-gazdag-x509-slhdsa.md

```diff
@@ -43,40 +54,434 @@ author:
+
+A SLH-DSA signature consists of the FORS signature, the WOTS+ signature in each layer, and the path to the root of each subtree until the root of the hypertree is reached.
+
+A SLH-DSA signature is verified by verifying the FORS signature, the WOTS+ signatures and the path to the root of each subtree.  When reaching the root of the hypertree, the signature verifies only if it hashes to the pre-trusted root of the SLH-DSA hypertree.
+
+SLH-DSA is a stateless hash-based signature algorithm.  Stateful hash-based signature schemes require that the WOTS+ private key (generated by using a state index) is never reused or the scheme loses it security.  Although its security decreases, FORS which is used at the bottom of the SLH-DSA hypertree does not collapse if the same private key used to sign two or more different messages like in stateful hash-based signature schemes.  Without the need for state kept by the signer to ensure it is not reused, SLH-DSA is much less fragile.
+
+SLH-DSA was designed to sign up to 2^64 messages and offers three security levels.  The parameters of the SLH-DSA hypertree include the security parameter, the hash function, the tree height, the number of layers of subtrees, the Winternitz parameter of WOTS+, the number of FORS trees and leaves in each.  The parameters for each of the security levels were chosen to provide 128 bits of security, 192 bits of security, and 256 bits of security.
+-->
+
+# Algorithm Identifiers {#sec-alg-ids}
+
+This specification uses placeholders for object identifiers until the identifiers for the new algorithms are assigned by NIST.
+
+The AlgorithmIdentifier type, which is included herein for convenience, is defined as follows:
+
+~~~
+    AlgorithmIdentifier  ::=  SEQUENCE  {
+        algorithm   OBJECT IDENTIFIER,
+        parameters  ANY DEFINED BY algorithm OPTIONAL
+    }
+
+    |  NOTE: The above syntax is from [RFC5280] and matches the
+    |  version used therein, i.e., the 1988 ASN.1 syntax.  See
+    |  [RFC5912] for ASN.1 copmatible with the 2015 ASN.1 syntax.
+~~~
+
+The fields in AlgorithmIdentifier have the following meanings:
+
+* algorithm identifies the cryptographic algorithm with an object identifier.
+
+* parameters, which are optional, are the associated parameters for the algorithm identifier in the algorithm field.
+
+The OIDs are:
```

Comment:
Did we decide to leave out SHA2-parameters for level III and V due to the discussion on the forum? Is it really decided in that way? If yes, I missed that.